keps/sig-storage/4049-storage-capacity-scoring-of-nodes-for-dynamic-provisioning/README.md

```diff
@@ -0,0 +1,970 @@
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Storage capacity needs to be considered when:
+
+- we want to resize after a node-local PV is scheduled. In this case we need to select a node with as much free space as possible. 
+- we want to select a node with less free node space to reduce the number of nodes as much as possible.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- To consider the amount of free node space on when performing dynamic provisioning.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- To change how to score nodes for static or dynamic provisioning.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+- Node scores based on available space can be taken into account when performing dynamic provisioning.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+We want to leave room for volume expansion after node allocation. In this case, we want to allocate the node that has the maximum amount of free space. 
+
+#### Story 2
+
+We want to reduce the number of nodes as much as possible to reduce costs when using a cloud environment. In this case, we want to allocate the node that has the smallest amount of sufficiently free space left.
```

Comment:
How does the score calculator know what node a Pod + PVC(s) want? The one with max. capacity or the least capacity? AFAIK `VolumeBindingArgs` are global for the whole scheduler, not per Pod/PVC

Reply:
> How does the score calculator know what node a Pod + PVC(s) want? The one with max. capacity or the least capacity?

The scoring logic is the same among all Pod + PVC(s). In other words, we can select (a) all Pod + PVC(s) prefer a node with the least capacity and (b) prefer a node with the maximum capacity. Otherwise, we can't achieve either (a) or (b).

> AFAIK `VolumeBindingArgs` are global for the whole scheduler, not per Pod/PVC

Users choose (a) or (b) in a global option in `VolumeBindingargs`.

Reply:
Please put it explicitly somewhere in the kep. Also, what is the default?

Reply:
Sure. I put it in the KEP.

> Also, what is the default?

In the case of local storage, I believe "Story 1" in the "User Stories" section is a common scenario. Therefore, I would like to set (b) (where nodes with larger capacity receive higher scores) as the default. I will also put this in the KEP.

Reply:
I have revised it.